File: AlgoritmoBurbujaLenguajes/AlgoritmoBurbuja.py
import numpy
from time import time
arreglo = list()

def llenarArreglo(N):
  aux=N
  for i in range(0,N,1):
    arreglo.insert(i,aux)
    aux-=1

def imprimirArreglo():
  for i in range(0,numpy.size(arreglo),1):
    print(arreglo[i])
def intercambio(min,i):
  aux=arreglo[i]
  arreglo[i]=arreglo[min]
  arreglo[min]=aux
  
def algoritmoBurbuja(N):
  start=time()
  for i in range(N,1,-1):
    for j in range(1,i,1):
      if arreglo[j-1]>arreglo[j]:
        intercambio(j-1,j)
  stop=time()
  elapsed_time=stop-start
  print("N: ",N,"Tiempo: ",elapsed_time,"s")
  #print(elapsed_time)

  


for i in range(100,10000,100):
  llenarArreglo(i)
  #imprimirArreglo()
  algoritmoBurbuja(i)
  #imprimirArreglo()

File: AlgoritmoBurbujaLenguajes/AlgoritmoBurbuja.cpp
#include <chrono>//libreria para calcular tiempos
#include <iostream>
using namespace std;
using namespace std::chrono;
using std::cout; using std::ofstream;
using std::endl; using std::string;
using std::fstream;
//metodo necesario para los intercambios en el ordenamiento burbuja
void intercambio(int a[], int min, int i){
	int aux;
	aux=a[i];
	a[i]=a[min];
	a[min]=aux;
}
//Metodo que ordena el arreglo usando el algoritmo burbuja
void burbuja (int a[], int N){
	int i,j;
	time_point<steady_clock> start, stop;
	start = steady_clock::now();
 	for (i=N; i>=1; i--)
		for(j=2; j<=i; j++)
 			if(a[j-1]> a[j]) intercambio(a,j-1,j);

 	stop = steady_clock::now();
 	duration<double> elapsed = (stop - start);
    cout<<"N:"<<N<<"\n"<<"Tiempo: "<<elapsed.count()<<" s."<<endl;
}

int main(int argc, char *argv[]){
	int N;
    cout<<"N representa la cantidad de datos en el arreglo"<<endl;
	for(int i=100; i<=10000; i+=100){
		N=i;
		int a[N];
		for(int j=N;j>=0;j--)
			a[N-j]=j;//Llena el arreglo en orden inverso, para que el algoritmo burbuja trabaje con su peor caso: O(N^2)
		burbuja(a, N);
	}

	return 0;
}
